lcs/agents/acs2/Classifier.py

```diff
@@ -84,10 +91,22 @@ def __hash__(self):
         return hash((str(self.condition), self.action, str(self.effect)))
 
     def __repr__(self):
-        return "{}-{}-{} @ {}".format(self.condition,
-                                      self.action,
-                                      self.effect,
-                                      hex(id(self)))
+        return "{} {:2} {:16} {:21} q: {:<5.3} r: {:<6.4} ir: {:<6.4} " \
+               "f: {:<6.4} exp: {:<3} tga: {:<5} talp: {:<5} tav: {:<6.3} " \
+               "num: {}".format(
+                    self.condition,
+                    gym_maze.ACTION_LOOKUP.get(self.action, '?'),
+                    str(self.effect),
+                    "(" + str(self.mark) + ")",
+                    float(self.q),
```

Comment:
Do we need casting here?

Reply:
I had trouble without casting, but they're gone with f-strings :)